lib/rubygems/resolver/molinillo/lib/molinillo/resolution.rb

```diff
@@ -713,8 +713,7 @@ def filtered_possibility_set(vertex)
       # @return [Object] the locked spec named `requirement_name`, if one
       #   is found on {#base}
       def locked_requirement_named(requirement_name)
-        vertex = base.vertex_named(requirement_name)
-        vertex && vertex.payload
+        base.vertex_named(requirement_name)&.payload
```

Comment:
Molinillo's files are vendored code, so we shouldn't modify them directly. You can propose the change [upstream](https://github.com/CocoaPods/Molinillo).